networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java

```diff
@@ -80,7 +83,12 @@ public LibP2PNetwork(
     this.privKey = config.getPrivateKey();
     this.nodeId = new LibP2PNodeId(PeerId.fromPubKey(privKey.publicKey()));
 
-    advertisedAddr = new Multiaddr("/ip4/127.0.0.1/tcp/" + config.getAdvertisedPort());
+    try {
+      advertisedAddr = composeAdvertisedAddr(config);
+    } catch (UnknownHostException err) {
```

Comment:
I'd probably move the exception handling to the `composeAdvertisedAddr` method

Reply:
done.